03week/ticTacToe.js

```diff
@@ -1,93 +1,223 @@
-'use strict';
+//code plan
+//convert functions to ES6
+//check if the placement is valid, then place it (tictactoe())
+//check for wins (checkForWin(), verticalWin(), horizontalWin(), diagonalWin(), isWinner())
+  //Each directional win function checks possible winner arrays against by looping through current board state(checkForWins()).  IsWinner() is used in the main game function to check for truthy return of direction check functions.
+//switchplayer each time main function is called (switchPlayer())
 
-const assert = require('assert');
-const readline = require('readline');
+"use strict";
+
+const assert = require("assert");
+const readline = require("readline");
 const rl = readline.createInterface({
   input: process.stdin,
   output: process.stdout
 });
 let board = [
-  [' ', ' ', ' '],
-  [' ', ' ', ' '],
-  [' ', ' ', ' ']
+  [" ", " ", " "], 
+  [" ", " ", " "], 
+  [" ", " ", " "]
 ];
 
-let playerTurn = 'X';
-
-function printBoard() {
-  console.log('   0  1  2');
-  console.log('0 ' + board[0].join(' | '));
-  console.log('  ---------');
-  console.log('1 ' + board[1].join(' | '));
-  console.log('  ---------');
-  console.log('2 ' + board[2].join(' | '));
-}
-
-function horizontalWin() {
-  // Your code here
-}
-
-function verticalWin() {
-  // Your code here
-}
-
-function diagonalWin() {
-  // Your code here
-}
-
-function checkForWin() {
-  // Your code here
+let playerTurn = "X";
+let gameover = false;
+let turn = 1;
+
+const printBoard= () => {
+  console.log("   0  1  2");
+  console.log("0 " + board[0].join(" | "));
+  console.log("  ---------");
+  console.log("1 " + board[1].join(" | "));
+  console.log("  ---------");
+  console.log("2 " + board[2].join(" | "));
 }
 
-function ticTacToe(row, column) {
-  // Your code here
+//Check for horizontal wins using arrays containing possible winning combos and checking against board state (checkForWin())
+const horizontalWin = () => {
```

Comment:
the function horizontalWin should return a boolean if there is a horizontal win, you shouldn't be calling checkForWin. checkForWin should return a truthy value if any of the win functions return true.